### ISSUE TITLE
VS Code extension is incomplete - does not show errors in some situations; cannot be configured from the UI

### ISSUE THREAD

**Issue by ffMathy** [oxc-project/oxc#2719]
The VS Code extension does not show any errors that `oxlint` finds when running it from the terminal.

In fact, what is the extension actually supposed to do? How do I know it is working?

**Comment by Boshen** [comment]
@IWANABETHATGUY Can you help a little bit with this one, we may need more debug prints and guides.

**Comment by vincentbriglia** [comment]
I think what you mean is that you cannot configure the vscode extension to enable or disable any rules 

you cannot mimic this functionality in vscode: ```npx oxlint@latest --tsconfig tsconfig.json -D all``` other than redefining a n ```.eslintrc.json``` file with the rules you want enabled. 

I really feel that this makes a case for an oxc-specific configuration file, even though there's a fundamental decision that this is a no-configuration library. 

Having looked at Biome, migration is harder from eslint to biome - if oxc would capitalize on its position where portability from eslint to Oxc is much simpler, and explicitely write to a configuration file, this tool would take off much more. 

I will write a vision statement for that

**Comment by Boshen** [comment]
VSCode support is incomplete and any help is appreciated.

We need:

* add configuration from the UI
* improve logging so we know what it's doing

**Comment by dalisoft** [comment]
Tracing with invalid configuration (see #3217 ) i got these errors

```
[2024-05-09T22:10:20Z INFO  oxc_language_server] initialize: Options { run: OnSave, enable: true, config_path: "eslint.config.js" }
[2024-05-09T22:10:20Z INFO  oxc_language_server] language server version: "0.0.1"
thread 'main' panicked at crates/oxc_language_server/src/main.rs:351:18:
should have initialized linter with new options:   x Failed to parse jsonc file "/Users/dalisoft/Desktop/ignore_folder/
  | test/test-backend/eslint.config.js"

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 3:10:20 AM] Server process exited with signal SIGABRT.
[Error - 3:10:20 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 3:10:20 AM] Connection to server got closed. Server will restart.
true
[Error - 3:10:20 AM] oxc client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[2024-05-09T22:10:20Z INFO  oxc_language_server] initialize: Options { run: OnSave, enable: true, config_path: "eslint.config.js" }
[2024-05-09T22:10:20Z INFO  oxc_language_server] language server version: "0.0.1"
thread 'main' panicked at crates/oxc_language_server/src/main.rs:351:18:
should have initialized linter with new options:   x Failed to parse jsonc file "/Users/dalisoft/Desktop/ignore_folder/
  | test/test-backend/eslint.config.js"

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 3:10:20 AM] Server process exited with signal SIGABRT.
[Error - 3:10:20 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 3:10:20 AM] Connection to server got closed. Server will restart.
true
[Error - 3:10:20 AM] oxc client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[2024-05-09T22:10:20Z INFO  oxc_language_server] initialize: Options { run: OnSave, enable: true, config_path: "eslint.config.js" }
[2024-05-09T22:10:20Z INFO  oxc_language_server] language server version: "0.0.1"
thread 'main' panicked at crates/oxc_language_server/src/main.rs:351:18:
should have initialized linter with new options:   x Failed to parse jsonc file "/Users/dalisoft/Desktop/ignore_folder/
  | test/test-backend/eslint.config.js"

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 3:10:20 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 3:10:20 AM] Connection to server got closed. Server will restart.
true
[Error - 3:10:20 AM] oxc client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 3:10:20 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 3:10:20 AM] Server process exited with signal SIGABRT.
[2024-05-09T22:10:20Z INFO  oxc_language_server] initialize: Options { run: OnSave, enable: true, config_path: "eslint.config.js" }
[2024-05-09T22:10:20Z INFO  oxc_language_server] language server version: "0.0.1"
thread 'main' panicked at crates/oxc_language_server/src/main.rs:351:18:
should have initialized linter with new options:   x Failed to parse jsonc file "/Users/dalisoft/Desktop/ignore_folder/
  | test/test-backend/eslint.config.js"

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 3:10:20 AM] Server process exited with signal SIGABRT.
[Error - 3:10:20 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 3:10:20 AM] Connection to server got closed. Server will restart.
true
[Error - 3:10:20 AM] oxc client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 3:10:20 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[2024-05-09T22:10:20Z INFO  oxc_language_server] initialize: Options { run: OnSave, enable: true, config_path: "eslint.config.js" }
[2024-05-09T22:10:20Z INFO  oxc_language_server] language server version: "0.0.1"
thread 'main' panicked at crates/oxc_language_server/src/main.rs:351:18:
should have initialized linter with new options:   x Failed to parse jsonc file "/Users/dalisoft/Desktop/ignore_folder/
  | test/test-backend/eslint.config.js"

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 3:10:20 AM] Server process exited with signal SIGABRT.
[Error - 3:10:20 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 3:10:20 AM] The oxc server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
[Error - 3:10:20 AM] oxc client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 3:10:20 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
```

**Comment by vincentbriglia** [comment]
currently only works with an `.eslintrc.json` file, oxlint doens't parse or resolve anything else. I think this is why it would be better to create an `oxlint.config.json` file for configuration to be less confusing. That way we can use that config in the eslint plugin with a clearer abstraction. 

then we could add a `configure` method config in `eslint-plugin-oxlint` f.ex: 

```js
export default {
  configs: {
    configure: (oxlintConfig: RuleConfigurations) => {
      const filteredConfig: RuleConfigurations = {};
      for (const key of Object.keys(oxlintConfig)) {
        if (Object.prototype.hasOwnProperty.call(allRules, key)) {
          filteredConfig[key] = oxlintConfig[key] as RuleSetting;
        } else {
          console.warn(
            `eslint-plugin-oxlint: Rule ${key} is not implemented in oxlint.`
          );
        }
      }
      return {
        name: 'oxlint',
        rules: filteredConfig,
      };
    },
    recommended: {
      plugins: ['oxlint'],
      rules: ruleMapsByCategory.correctnessRules,
    },
    ...
}
```

what do you think @Boshen ?

**Comment by NateRadebaugh** [comment]
Thanks so much for this! In testing this, I noticed a couple other missing features from the vscode extension:
* Showing the actual lint error name in the UI instead of just the error to make for easy investigation/disabling
* Ctrl+click in the terminal will navigate to the file, but not the specific line the issue is on

**Comment by IWANABETHATGUY** [comment]
> Thanks so much for this! In testing this, I noticed a couple other missing features from the vscode extension:
> 
> * Showing the actual lint error name in the UI instead of just the error to make for easy investigation/disabling
> * Ctrl+click in the terminal will navigate to the file, but not the specific line the issue is on

for the first one: tracked here https://github.com/oxc-project/oxc/issues/6477
for the second one, can you elaborate? Sounds like it related to `oxlint` rather than the `vscode` extension.

**Comment by NateRadebaugh** [comment]
> > Thanks so much for this! In testing this, I noticed a couple other missing features from the vscode extension:
> > 
> > * Showing the actual lint error name in the UI instead of just the error to make for easy investigation/disabling
> > * Ctrl+click in the terminal will navigate to the file, but not the specific line the issue is on
> 
> for the first one: tracked here #6477 for the second one, can you elaborate? Sounds like it related to `oxlint` rather than the `vscode` extension.

Thanks I'll follow that one.

As far as file linking in the vscode terminal, here's the docs I found about that:
https://code.visualstudio.com/docs/terminal/basics#_links

It looks like there's a standard format it's expecting but the oxlint output does not match.

**Comment by nrayburn-tech** [comment]
It would be nice to have links work within IntelliJ IDEs as well.  I haven't been able to find any docs on what is supported though.

I think https://youtrack.jetbrains.com/issue/IJPL-103639/Terminal-toolwindow-should-linkify-references-to-project-sources is the feature request.  https://youtrack.jetbrains.com/issue/IJPL-109865/Make-opening-external-files-from-the-embedded-terminal-configurable indicates that file URLs (file://) are supported, but nothing else is mentioned.

**Comment by nrayburn-tech** [comment]
@NateRadebaugh it seems like links work for me (video below).  If you still have an issue with links in the editor, could you create a new issue with some steps to reproduce?

https://github.com/user-attachments/assets/27bffe58-c9e8-4d9b-bb5f-fa4f0d7811d5

**Comment by nrayburn-tech** [comment]
At this point, I think this issue can probably be closed and new issues created for any specific issues.

General list of available features below.
- Language server path can be specified. `oxc.path.server`.
- Config file can be specified `oxc.configPath`.
- Language server can be enabled/disabled `oxc.enable` (not really sure what this is useful for).
- Running on save or lint can be configured `oxc.lint.run`.
- `oxc.trace.server` can be used to enable verbose logs between the language server and language client.
- Oxlint config files should have the correct JSON schema attached (ESLint config files don't have the Oxlint schema, maybe that's something worth changing)
    - oxlintrc.json
    - oxlint.json
    - .oxlintrc.json
    - .oxlint.json
    - .oxlintrc
- When opening a file that is javascript, javascriptreact, typescript, typescriptreact, vue, or svelte the VS Code extension will activate.
- There is an `Oxc` channel for viewing logs.
- There is a command available to restart the language server.
- Highlighting is shown in the editor for diagnostics published by the language server.
